components/eamxx/src/physics/p3/eamxx_p3_process_interface.hpp

```diff
@@ -361,7 +361,11 @@ class P3Microphysics : public AtmosphereProcess
     // 1d view scalar, size (ncol)
     static constexpr int num_1d_scalar = 2; //no 2d vars now, but keeping 1d struct for future expansion
     // 2d view packed, size (ncol, nlev_packs)
+#ifdef SCREAM_P3_SMALL_KERNELS
+    static constexpr int num_2d_vector = 64;
```

Comment:
I suppose at some point we should try to see if all these views need to be different, or if we can have a few that alias each other. I guess there are a few that can alias each other, but knowing which ones would require an analysis of p3. Also, allowing aliasing put some constraints on the small kernel execution order, since rearranging them may cause data to be not what was expected.

Still, food for thoughts.

Reply:
Yes, almost certainly there could be some reduction. I'll make an issue.